### PR TITLE
[WIP] Documentation updates

### PR DESCRIPTION
**Pull request by quietlychris**
Adds initial documentation of `linfa-bayes` subcrate, small edit for `linfa_clustering`

### CODE REVIEW

**Comment by bytesnake** [comment]
ping me when you want a review :)

**Comment by quietlychris** [comment]
Will do! I'm hoping to get through a couple more bullet points on the list from #82 a little later today, and then will let you know.

**Comment by quietlychris** [comment]
@bytesnake I think this should be ready for a review. Right now, we're not passing a clippy lint (odd, because I didn't change any code...) but I can take care of that if you'd like me to before merging into `master`. This isn't a huge contribution by any means, but hopefully it helps a little bit. I'll keep trying to find some time to add stuff like this. 

Of course, happy to make any changes you think are necessary!

**Comment by bytesnake** [comment]
can you rebase your PR to the master branch? Clippy added an additional rule recently, that `filter_map(|x| x)` can be replaced by `flatten()`and fails therefore but passed previously

**Comment by quietlychris** [comment]
@bytesnake I think this should do it. Not sure if it actually rebased on `master` because I had to manually make the change (the fix wasn't showing up in my local branch even when pulling), but the CI seems happy with it

**Comment by bytesnake** [comment]
sorry these were two separate sentences :laughing: can you do the rebase to master as well so that the stall commits are removed from the history

**Comment by quietlychris** [comment]
Ahhhhhh. This got closed accidentally during an attempted re-base which went horribly wrong.

File: linfa-bayes/src/gaussian_nb.rs [hunk 1]
```diff
@@ -13,7 +7,11 @@ use linfa::dataset::{DatasetBase, Labels};
 use linfa::traits::{Fit, IncrementalFit, Predict};
 use linfa::Float;
 
-/// Gaussian Naive Bayes (GaussianNB)
+/// Gaussian Naive Bayes (GaussianNB) hyperparameters
+///
+/// The Gaussian Naive Bayes is a classification algorithm where the likelihood
+/// of the feature P(x_i | y) is assumed to be Gaussian, and the mean and variance
+/// are estimated using maximum likelihood.
```
Comment (bytesnake) [inline linfa-bayes/src/gaussian_nb.rs:14]: perhaps also: "and features are assumed to be independent."

**Review by bytesnake — COMMENTED** [review]
git is sometimes tough, there are still two merge conflicts :sweat_smile: the changes are looking otherwise good to me

**Pull request closed without merging**